docs/manuals/source/DeveloperGuide/release.rst

```diff
@@ -0,0 +1,74 @@
+Releasing Bareos
+----------------
+
+This chapter describes how to release a new version of Bareos.
+The documentation is limited to the changes that need to be done to the sources and the git repository.
+Building and distributing packages is not the scope of this chapter.
+
+.. note::
+   Consider this a step-by-step manual. The sections are meant to be carried out in that order.
+
+Deciding for a version number
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+When you release you will usually need at least two version numbers: the version you're going to release and the version that will follow after that version.
+For details on the numbering-scheme, take a look at :ref:`version-numbers`.
+
+Usually, you will have an existing work-in-progress tag that generates pre-release version numbers for your branch.
+So the version you're going to release is probably the final version for that pre-release.
+
+Preparing the release notes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+For each new release there should be a section in the :ref:`releasenotes`.
+You can use the release notes of a previous release as a template.
+
+The release notes should be committed to the master branch and then cherry-picked to your release branch using ``git cherry-pick -x``.
+
+Update version-dependent files
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+There are a few version-dependent files in the Bareos sources that need to be changed.
+
+version.h
+^^^^^^^^^
+Starting with Bareos :sinceVersion:`19:remove version.h` the values for ``version.h`` are automatically determined by cmake based on git repository data.
+Changing ``version.h`` in such a new release is not neccessary anymore.
+
```

Comment:
I've seen newly added code like
```
cmake -P write_version_files.cmake
```
which I consider related.When is it required? Would a remark about it make sense here or better at https://docs.bareos.org/master/DeveloperGuide/BuildAndTestBareos.html ?

Reply:
You need to call only when you create tarballs.
Preparing tarballs is not covered in that chapter, I might need to rephrase the introduction to make this clearer.

Reply:
We also need to make clear that "Build and Test Bareos" doesn't tell people to build in the core-Directory (which will fail unless you called write_version_files.cmake first).